### PR TITLE
Dismiss keyboard onscroll

### PR DESCRIPTION
**Pull request by Apsaliya**
#15

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -84,6 +84,7 @@ buildscript {
           'android': 'io.reactivex.rxjava2:rxandroid:2.0.1',
           'binding': [
               'framework': 'com.jakewharton.rxbinding2:rxbinding-kotlin:2.1.0',
```
Comment (jaredsburrows) [inline build.gradle]: Can you add a constant for `2.1.0`?

File: frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt [hunk 2]
```diff
@@ -154,6 +158,21 @@ class MainActivity : Activity() {
...
+    val vc = ViewConfiguration.get(recycler.context)
+    val slop = vc.scaledTouchSlop
+    recycler.scrollEvents()
+            .filter({ scroll -> scroll.dy() > 0 })
+            .observeOn(mainThread())
+            .crashingSubscribe {
+              scroll -> totalDy += scroll.dy()
+              if (totalDy >= slop) {
+                val inputManager = applicationContext.getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
+                inputManager.hideSoftInputFromWindow(currentFocus!!.windowToken, HIDE_NOT_ALWAYS)
```
Comment (jaredsburrows) [inline frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt]: Can you make `!!.` be `?.`?

File: build.gradle (hunk 1 above)
Comment (Apsaliya) [reply build.gradle]: done.

File: frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt (hunk 2 above)
Comment (Apsaliya) [reply frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt]: done.

File: frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt [hunk 3]
```diff
@@ -154,6 +158,21 @@ class MainActivity : Activity() {
...
+    var totalDy = 0
+    val vc = ViewConfiguration.get(recycler.context)
+    val slop = vc.scaledTouchSlop
+    recycler.scrollEvents()
+            .filter({ scroll -> scroll.dy() > 0 })
+            .observeOn(mainThread())
+            .crashingSubscribe {
+              scroll -> totalDy += scroll.dy()
+              if (totalDy >= slop) {
+                val inputManager = applicationContext.getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
```
Comment (jshvarts) [inline frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt]: How about using `systemService` extension function found in utils?
```
    systemService<InputMethodManager>().apply {
        hideSoftInputFromWindow(currentFocus?.windowToken, HIDE_NOT_ALWAYS)
    }
```

Comment (Apsaliya) [reply frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt]: Cool! Didn't know about this. I'll update this.

File: frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt [hunk 4]
```diff
@@ -154,6 +159,22 @@ class MainActivity : Activity() {
           queryInput.typeface = if (empty) Typeface.DEFAULT else robotoMono
         }
 
+    var totalDy = 0
+    val vc = ViewConfiguration.get(recycler.context)
+    val slop = vc.scaledTouchSlop
+    recycler.scrollEvents()
+            .filter({ scroll -> scroll.dy() > 0 })
```
Comment (JakeWharton) [inline frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt]: remove parenthesis

File: frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt [hunk 5]
```diff
@@ -154,6 +159,22 @@ class MainActivity : Activity() {
...
+            .observeOn(mainThread())
```
Comment (JakeWharton) [inline frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt]: delete, events are already emitted on the main thread

File: frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt (hunk 4 above)
Comment (JakeWharton) [inline frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt]: indent should be +4 not +8

File: frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt [hunk 6]
```diff
@@ -154,6 +159,22 @@ class MainActivity : Activity() {
...
+            .crashingSubscribe {
+              scroll -> totalDy += scroll.dy()
```
Comment (JakeWharton) [inline frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt]: lambda argument name and arrow should be after `{`

File: frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt [hunk 7]
```diff
@@ -154,6 +159,22 @@ class MainActivity : Activity() {
...
+              if (totalDy >= slop) {
+                systemService<InputMethodManager>().apply {
```
Comment (JakeWharton) [inline frontend/android/src/main/java/com/jakewharton/sdksearch/ui/MainActivity.kt]: remove `.apply`, it's useless. just call the method on the returned value directly

**Comment by JakeWharton** [comment]
This works great, thanks! Squashed and merged.

**Pull request closed without merging**